### PR TITLE
feat(permissions): Slash Permissions v2

### PR DESCRIPTION
**Pull request by coolcalcacol**
Remove bulk editing of command permissions as the endpoint has been disabled, 
add defaultMemberPermissions and dmPermission in place of defaultPermission, 
add the applicationCommandPermissionsUpdate gateway event, 
add the ApplicationCommandPermissionTypes channel type, 
modify createGuildCommand to return without the new DMPermission field

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -175,7 +175,8 @@ declare namespace Eris {
   // Application Commands
   interface ApplicationCommand<T extends ApplicationCommandTypes = ApplicationCommandTypes> {
     application_id: string;
-    defaultPermission?: boolean;
+    defaultMemberPermissions?: string;
+    DMPermission?: boolean;
```
Comment (DonovanDMC) [inline index.d.ts]:
```suggestion
    dmPermission?: boolean;
```
keys should be camelCased

File: index.d.ts [hunk 2]
```diff
@@ -2234,7 +2236,7 @@ declare namespace Eris {
     createCommand(command: ApplicationCommandStructure): Promise<ApplicationCommand>;
     createGroupChannel(userIDs: string[]): Promise<GroupChannel>;
     createGuild(name: string, options?: CreateGuildOptions): Promise<Guild>;
-    createGuildCommand(guildID: string, command: ApplicationCommandStructure): Promise<ApplicationCommand>;
+    createGuildCommand(guildID: string, command: Omit<ApplicationCommandStructure, 'DMPermission'>): Promise<Omit<ApplicationCommand, 'DMPermission'>>;
```
Comment (DonovanDMC) [inline index.d.ts]:
```suggestion
    createGuildCommand(guildID: string, command: Omit<ApplicationCommandStructure, 'dmPermission'>): Promise<Omit<ApplicationCommand, 'dmPermission'>>;
```
keys should be camel cased

File: lib/Client.js [hunk 3]
```diff
@@ -566,7 +552,10 @@ class Client extends EventEmitter {
                 }
             }
         }
-        command.default_permission = command.defaultPermission;
+
+        command.default_member_permissions = command.defaultMemberPermissions;
+        command.dm_permission = command.DMPermission;
```
Comment (DonovanDMC) [inline lib/Client.js]:
```suggestion
        command.dm_permission = command.dmPermission;
```
keys should be camel cased

File: lib/Client.js [hunk 4]
```diff
@@ -554,7 +539,8 @@ class Client extends EventEmitter {
     * @arg {String} [command.description] The command description (Slash Commands Only)
     * @arg {Array<Object>} [command.options] An array of [command options](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure)
     * @arg {Number} [type=1] The type of application command, 1 for slash command, 2 for user, and 3 for message
-    * @arg {Boolean} [command.defaultPermission=true] Whether the command is enabled by default when the app is added to a guild
+    * @arg {String} [command.defaultMemberPermissions] The default member permissions represented as a
+    * @arg {Boolean} [command.DMPermission] The default role permissions
```
Comment (DonovanDMC) [inline lib/Client.js]:
```suggestion
    * @arg {Boolean} [command.dmPermission] The default role permissions
```
keys should be camel cased

**Review by DonovanDMC — CHANGES_REQUESTED** [review]
keys should be camel cased

File: lib/gateway/Shard.js [hunk 5]
```diff
@@ -2337,6 +2337,15 @@ class Shard extends EventEmitter {
                 this.emit("interactionCreate", Interaction.from(packet.d, this.client));
                 break;
             }
+            case "APPLICATION_COMMAND_PERMISSIONS_UPDATE": {
+                /**
+                 * Fired when an application command's permissions are updated
+                 * @event Client#applicationCommandPermissionsUpdate
+                 * @prop {GuildApplicationCommandPermissions} GuildApplicationCommandPermissions The updated command permissions
```
Comment (bsian03) [inline lib/gateway/Shard.js]: event prop names should be camelCase

**Review by bsian03 — DISMISSED** [review]

File: index.d.ts [hunk 6]
```diff
@@ -2234,7 +2236,7 @@ declare namespace Eris {
     createCommand(command: ApplicationCommandStructure): Promise<ApplicationCommand>;
     createGroupChannel(userIDs: string[]): Promise<GroupChannel>;
     createGuild(name: string, options?: CreateGuildOptions): Promise<Guild>;
-    createGuildCommand(guildID: string, command: ApplicationCommandStructure): Promise<ApplicationCommand>;
+    createGuildCommand(guildID: string, command: Omit<ApplicationCommandStructure, 'dmPermission'>): Promise<Omit<ApplicationCommand, 'dmPermission'>>;
```
Comment (DonovanDMC) [inline index.d.ts]:
```suggestion
    createGuildCommand(guildID: string, command: Omit<ApplicationCommandStructure, "dmPermission">): Promise<Omit<ApplicationCommand, "dmPermission">>;
```

**Review by DonovanDMC — CHANGES_REQUESTED** [review]
jsdoc hasn't been changed for `Client#createGuildCommand`

**Review by DonovanDMC — CHANGES_REQUESTED** [review]
`bulkEditCommands`/`bulkEditGuildCommands` does not properly replace the camel cased names with underscored names

**Review by DonovanDMC — CHANGES_REQUESTED** [review]
`editCommand` / `editGuildCommand` also have not been modified

File: index.d.ts [hunk 7]
```diff
@@ -682,6 +683,7 @@ declare namespace Eris {
     selfVideo: boolean;
   }
   interface EventListeners {
+    applicationCommandPermissionsUpdate: [applicationCommandPermissions: GuildApplicationCommandPermissions]
```
Comment (DonovanDMC) [inline index.d.ts]:
```suggestion
    applicationCommandPermissionsUpdate: [applicationCommandPermissions: GuildApplicationCommandPermissions];
```

File: lib/Client.js [hunk 8]
```diff
@@ -554,19 +542,23 @@ class Client extends EventEmitter {
...
                 if(!command.name.match(/^[\w-]{1,32}$/)) {
                     throw new Error("Slash Command names must match the regular expression \"^[\\w-]{1,32}$\"");
                 }
             }
         }
-        command.default_permission = command.defaultPermission;
+
+        command.default_member_permissions = command.defaultMemberPermissions;
+        command.dm_permission = command.dmPermission;
+
```
Comment (DonovanDMC) [inline lib/Client.js]:
```suggestion
        command.default_member_permissions = command.defaultMemberPermissions;
        command.dm_permission = command.dmPermission;
```

File: lib/Client.js [hunk 9]
```diff
@@ -624,19 +616,21 @@ class Client extends EventEmitter {
...
                 command.name = command.name.toLowerCase();
                 if(!command.name.match(/^[\w-]{1,32}$/)) {
                     throw new Error("Slash Command names must match the regular expression \"^[\\w-]{1,32}$\"");
                 }
             }
         }
-        command.default_permission = command.defaultPermission;
+
+        command.default_member_permissions = command.defaultMemberPermissions;
+
```
Comment (DonovanDMC) [inline lib/Client.js]:
```suggestion
        command.default_member_permissions = command.defaultMemberPermissions;
```

**Review by DonovanDMC — CHANGES_REQUESTED** [review]
consistent spacing (github isn't displaying it, but the newline after should also be removed)

**Review by DonovanDMC — DISMISSED** [review]

File: lib/Client.js [hunk 10]
```diff
@@ -640,19 +630,19 @@ class Client extends EventEmitter {
     * @arg {String} [command.description] The command description (Slash Commands Only)
     * @arg {Array<Object>} [command.options] An array of [command options](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure)
     * @arg {Number} [type=1] The type of application command, 1 for slash command, 2 for user, and 3 for message
-    * @arg {Boolean} [command.defaultPermission] Whether the command is enabled by default when the app is added to a guild
+    * @arg {String?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```
Comment (bsian03) [inline lib/Client.js]: As below

File: lib/Client.js [hunk 11]
```diff
@@ -1380,22 +1370,24 @@ class Client extends EventEmitter {
     * @arg {String} command.name The command name
     * @arg {String} [command.description] The command description (Slash Commands Only)
     * @arg {Array<Object>} [command.options] An array of [command options](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure)
-    * @arg {Boolean} [command.defaultPermission] Whether the command is enabled by default when the app is added to a guild
-    * @returns {Promise<Object>} Resolves with a commands object
+    * @arg {String?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```
Comment (bsian03) [inline lib/Client.js]: As above

File: lib/Client.js [hunk 12]
```diff
@@ -1473,22 +1465,22 @@ class Client extends EventEmitter {
     * @arg {String} command.name The command name
     * @arg {String} [command.description] The command description (Slash Commands Only)
     * @arg {Array<Object>} [command.options] An array of [command options](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure)
-    * @arg {Boolean} [command.defaultPermission] Whether the command is enabled by default when the app is added to a guild
+    * @arg {String?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```
Comment (bsian03) [inline lib/Client.js]: As below

File: lib/structures/Guild.js [hunk 13]
```diff
@@ -411,7 +411,7 @@ class Guild extends Base {
     * @arg {String} [command.description] The command description (Slash Commands Only)
     * @arg {Array<Object>} [command.options] An array of [command options](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure)
     * @arg {Number} [type=1] The type of application command, 1 for slash command, 2 for user, and 3 for message
-    * @arg {Boolean} [command.defaultPermission] Whether the command is enabled by default when the app is added to a guild
+    * @arg {String?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```
Comment (bsian03) [inline lib/structures/Guild.js]: As below

File: lib/structures/Guild.js [hunk 14]
```diff
@@ -637,7 +637,7 @@ class Guild extends Base {
     * @arg {String} command.name The command name
     * @arg {String} [command.description] The command description (Slash Commands Only)
     * @arg {Array<Object>} [command.options] An array of [command options](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure)
-    * @arg {Boolean} [command.defaultPermission] Whether the command is enabled by default when the app is added to a guild
+    * @arg {String?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```
Comment (bsian03) [inline lib/structures/Guild.js]: As below

File: lib/Client.js [hunk 15]
```diff
@@ -570,19 +558,21 @@ class Client extends EventEmitter {
     * @arg {String} [command.description] The command description (Slash Commands Only)
     * @arg {Array<Object>} [command.options] An array of [command options](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure)
     * @arg {Number} [type=1] The type of application command, 1 for slash command, 2 for user, and 3 for message
-    * @arg {Boolean} [command.defaultPermission=true] Whether the command is enabled by default when the app is added to a guild
-    * @returns {Promise<Object>} Resolves with a commands object
+    * @arg {String?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```
Comment (bsian03) [inline lib/Client.js]: As below

File: index.d.ts [hunk 16]
```diff
@@ -175,7 +175,8 @@ declare namespace Eris {
   // Application Commands
   interface ApplicationCommand<T extends ApplicationCommandTypes = ApplicationCommandTypes> {
     application_id: string;
-    defaultPermission?: boolean;
+    defaultMemberPermissions?: string;
```
Comment (bsian03) [inline index.d.ts]: Should be able to pass BigInt/Number/String/Permission to be consistent with other permission related methods elsewhere

**Review by bsian03 — CHANGES_REQUESTED** [review]
Audit log handling missing

File: lib/Client.js [hunk 17]
```diff
@@ -353,36 +353,26 @@ class Client extends EventEmitter {
...
+            if(command.name !== undefined) {
                 if(command.type === 1 || command.type === undefined) {
                     command.name = command.name.toLowerCase();
                     if(!command.name.match(/^[\w-]{1,32}$/)) {
                         throw new Error("Slash Command names must match the regular expression \"^[\\w-]{1,32}$\"");
                     }
                 }
             }
+            if(command.defaultMemberPermissions !== undefined) {
+                command.defaultMemberPermissions = command.defaultMemberPermissions instanceof Permission ? String(command.defaultMemberPermissions.allow) : String(command.defaultMemberPermissions);
```
Comment (bsian03) [inline lib/Client.js]: This seems kinda useless

File: lib/Client.js [hunk 18]
```diff
@@ -395,14 +385,18 @@ class Client extends EventEmitter {
...
```
Comment (bsian03) [inline lib/Client.js]: Same with here

File: lib/Client.js [hunk 19]
```diff
@@ -570,19 +564,24 @@ class Client extends EventEmitter {
     * @arg {String} [command.description] The command description (Slash Commands Only)
     * @arg {Array<Object>} [command.options] An array of [command options](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure)
     * @arg {Number} [type=1] The type of application command, 1 for slash command, 2 for user, and 3 for message
-    * @arg {Boolean} [command.defaultPermission=true] Whether the command is enabled by default when the app is added to a guild
-    * @returns {Promise<Object>} Resolves with a commands object
+    * @arg {bigint | number | string | Permission?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```
Comment (bsian03) [inline lib/Client.js]:
```suggestion
    * @arg {BigInt | Number | String | Permission?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```

File: lib/Client.js [hunk 20]
```diff
@@ -570,19 +564,24 @@ class Client extends EventEmitter {
...
+    * @arg {Boolean?} [command.dmPermission] Indicates whether the command is available in DMs with the app. By default, commands are visible.
+     * @returns {Promise<Object>} Resolves with a commands object
```
Comment (bsian03) [inline lib/Client.js]:
```suggestion
    * @returns {Promise<Object>} Resolves with a commands object
```

File: lib/Client.js [hunk 21]
```diff
@@ -570,19 +564,24 @@ class Client extends EventEmitter {
...
             if(command.type === 1 || command.type === undefined) {
                 command.name = command.name.toLowerCase();
                 if(!command.name.match(/^[\w-]{1,32}$/)) {
                     throw new Error("Slash Command names must match the regular expression \"^[\\w-]{1,32}$\"");
                 }
             }
         }
-        command.default_permission = command.defaultPermission;
+        if(command.defaultMemberPermissions !== undefined) {
+            command.defaultMemberPermissions = command.defaultMemberPermissions instanceof Permission ? String(command.defaultMemberPermissions.allow) : String(command.defaultMemberPermissions);
```
Comment (bsian03) [inline lib/Client.js]: Same here

File: lib/Client.js [hunk 22]
```diff
@@ -640,19 +639,22 @@ class Client extends EventEmitter {
     * @arg {String} [command.description] The command description (Slash Commands Only)
     * @arg {Array<Object>} [command.options] An array of [command options](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure)
     * @arg {Number} [type=1] The type of application command, 1 for slash command, 2 for user, and 3 for message
-    * @arg {Boolean} [command.defaultPermission] Whether the command is enabled by default when the app is added to a guild
+    * @arg {bigint | number | string | Permission?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```
Comment (bsian03) [inline lib/Client.js]:
```suggestion
    * @arg {BigInt | Number | String | Permission?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```

File: lib/Client.js [hunk 23]
```diff
@@ -640,19 +639,22 @@ class Client extends EventEmitter {
...
             if(command.type === 1 || command.type === undefined) {
                 command.name = command.name.toLowerCase();
                 if(!command.name.match(/^[\w-]{1,32}$/)) {
                     throw new Error("Slash Command names must match the regular expression \"^[\\w-]{1,32}$\"");
                 }
             }
         }
-        command.default_permission = command.defaultPermission;
+        if(command.defaultMemberPermissions !== undefined) {
+            command.defaultMemberPermissions = command.defaultMemberPermissions instanceof Permission ? String(command.defaultMemberPermissions.allow) : String(command.defaultMemberPermissions);
```
Comment (bsian03) [inline lib/Client.js]: Same here

File: lib/Client.js [hunk 24]
```diff
@@ -1380,22 +1382,27 @@ class Client extends EventEmitter {
     * @arg {String} command.name The command name
     * @arg {String} [command.description] The command description (Slash Commands Only)
     * @arg {Array<Object>} [command.options] An array of [command options](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure)
-    * @arg {Boolean} [command.defaultPermission] Whether the command is enabled by default when the app is added to a guild
-    * @returns {Promise<Object>} Resolves with a commands object
+    * @arg {bigint | number | string | Permission?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```
Comment (bsian03) [inline lib/Client.js]:
```suggestion
    * @arg {BigInt | Number | String | Permission?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```

File: lib/Client.js [hunk 25]
```diff
@@ -1380,22 +1382,27 @@ class Client extends EventEmitter {
...
+    * @arg {Boolean?} [command.dmPermission] Indicates whether the command is available in DMs with the app. By default, commands are visible.
+     * @returns {Promise<Object>} Resolves with a commands object
```
Comment (bsian03) [inline lib/Client.js]:
```suggestion
    * @returns {Promise<Object>} Resolves with a commands object
```

File: lib/Client.js [hunk 26]
```diff
@@ -1380,22 +1382,27 @@ class Client extends EventEmitter {
...
             if(command.type === 1 || command.type === undefined) {
                 command.name = command.name.toLowerCase();
                 if(!command.name.match(/^[\w-]{1,32}$/)) {
                     throw new Error("Slash Command names must match the regular expression \"^[\\w-]{1,32}$\"");
                 }
             }
         }
-        command.default_permission = command.defaultPermission;
+        if(command.defaultMemberPermissions !== undefined) {
+            command.defaultMemberPermissions = command.defaultMemberPermissions instanceof Permission ? String(command.defaultMemberPermissions.allow) : String(command.defaultMemberPermissions);
```
Comment (bsian03) [inline lib/Client.js]: Same here

File: lib/Client.js [hunk 27]
```diff
@@ -1473,22 +1480,25 @@ class Client extends EventEmitter {
     * @arg {String} command.name The command name
     * @arg {String} [command.description] The command description (Slash Commands Only)
     * @arg {Array<Object>} [command.options] An array of [command options](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-structure)
-    * @arg {Boolean} [command.defaultPermission] Whether the command is enabled by default when the app is added to a guild
+    * @arg {bigint | number | string | Permission?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```
Comment (bsian03) [inline lib/Client.js]:
```suggestion
    * @arg {BigInt | Number | String | Permission?} [command.defaultMemberPermissions] The default member [permissions](https://discord.com/developers/docs/topics/permissions) represented as a bit set
```

File: lib/Client.js [hunk 28]
```diff
@@ -1473,22 +1480,25 @@ class Client extends EventEmitter {
...
             if(command.type === 1 || command.type === undefined) {
                 command.name = command.name.toLowerCase();
                 if(!command.name.match(/^[\w-]{1,32}$/)) {
                     throw new Error("Slash Command names must match the regular expression \"^[\\w-]{1,32}$\"");
                 }
             }
         }
-        command.default_permission = command.defaultPermission;
+        if(command.defaultMemberPermissions !== undefined) {
+            command.defaultMemberPermissions = command.defaultMemberPermissions instanceof Permission ? String(command.defaultMemberPermissions.allow) : String(command.defaultMemberPermissions);
```
Comment (bsian03) [inline lib/Client.js]: Same here

File: lib/gateway/Shard.js [hunk 29]
```diff
@@ -2338,11 +2338,20 @@ class Shard extends EventEmitter {
...
+                * @prop {PingInteraction | CommandInteraction | ComponentInteraction | AutocompleteInteraction | UnknownInteraction} interaction The Interaction that was created
                 */
                 this.emit("interactionCreate", Interaction.from(packet.d, this.client));
                 break;
             }
+            case "APPLICATION_COMMAND_PERMISSIONS_UPDATE": {
+                /**
+                 * Fired when an application command's permissions are updated
+                 * @event Client#applicationCommandPermissionsUpdate
+                 * @prop {GuildApplicationCommandPermissions} guildApplicationCommandPermissions The updated command permissions
```
Comment (bsian03) [inline lib/gateway/Shard.js]:
```suggestion
                 * @prop {Object} guildApplicationCommandPermissions The updated command permissions
```
Can't use typescript interfaces in JSDoc

**Review by bsian03 — CHANGES_REQUESTED** [review]

File: lib/structures/Guild.js (hunk 14 above)
Comment (bsian03) [reply lib/structures/Guild.js]: This wasn't updated

File: lib/structures/Guild.js (hunk 13 above)
Comment (bsian03) [reply lib/structures/Guild.js]: This wasn't updated

File: index.d.ts [hunk 30]
```diff
@@ -175,7 +175,8 @@ declare namespace Eris {
   // Application Commands
   interface ApplicationCommand<T extends ApplicationCommandTypes = ApplicationCommandTypes> {
     application_id: string;
-    defaultPermission?: boolean;
```
Comment (bsian03) [inline index.d.ts:186]: This should be retained for deprecation in line with Discord

File: index.d.ts [hunk 31]
```diff
@@ -175,7 +175,8 @@ declare namespace Eris {
...
+    defaultMemberPermissions?: bigint | number | string | Permission;
+    dmPermission?: boolean;
```
Comment (bsian03) [inline index.d.ts]: ~~These shold be snake_case, responses from Discord are being treated as is. Not sure why default_permission wasn't treated that way though...~~

File: index.d.ts [hunk 32]
```diff
@@ -694,6 +695,7 @@ declare namespace Eris {
     selfVideo: boolean;
   }
   interface EventListeners {
+    applicationCommandPermissionsUpdate: [applicationCommandPermissions: GuildApplicationCommandPermissions];
```
Comment (bsian03) [inline index.d.ts:730]: L256 `permissions` is a guaranteed property

File: lib/structures/Guild.js
Comment (bsian03) [inline lib/structures/Guild.js:1]: bulkEditCommandPermissions needs removing

File: index.d.ts [hunk 33]
```diff
@@ -2258,7 +2260,7 @@ declare namespace Eris {
     createCommand(command: ApplicationCommandStructure): Promise<ApplicationCommand>;
     createGroupChannel(userIDs: string[]): Promise<GroupChannel>;
     createGuild(name: string, options?: CreateGuildOptions): Promise<Guild>;
-    createGuildCommand(guildID: string, command: ApplicationCommandStructure): Promise<ApplicationCommand>;
+    createGuildCommand(guildID: string, command: Omit<ApplicationCommandStructure, "dmPermission">): Promise<Omit<ApplicationCommand, "dmPermission">>;
```
Comment (bsian03) [inline index.d.ts]: Should be `dm_permission`, see other review

File: index.d.ts [hunk 34]
```diff
@@ -175,7 +175,8 @@ declare namespace Eris {
   // Application Commands
   interface ApplicationCommand<T extends ApplicationCommandTypes = ApplicationCommandTypes> {
     application_id: string;
-    defaultPermission?: boolean;
+    defaultMemberPermissions?: bigint | number | string | Permission;
```
Comment (bsian03) [inline index.d.ts]: In incoming data, this is a nullable string only. In outgoing data, this is as handled in js code, but should also be nullable

File: lib/Client.js (hunk 17 above)
Comment (bsian03) [reply lib/Client.js]: Should allow for nullable handling

File: lib/Client.js [hunk 35]
```diff
@@ -353,36 +353,26 @@ class Client extends EventEmitter {
...
                     command.name = command.name.toLowerCase();
                     if(!command.name.match(/^[\w-]{1,32}$/)) {
                         throw new Error("Slash Command names must match the regular expression \"^[\\w-]{1,32}$\"");
                     }
                 }
             }
+            if(command.defaultMemberPermissions !== undefined) {
+                command.defaultMemberPermissions = command.defaultMemberPermissions instanceof Permission ? String(command.defaultMemberPermissions.allow) : String(command.defaultMemberPermissions);
+            }
+            command.default_member_permissions = command.defaultMemberPermissions;
```
Comment (bsian03) [inline lib/Client.js]: This can be put in the unconditional check

**Review by bsian03 — CHANGES_REQUESTED** [review]
Starting to consider whether its worth having a separate ApplicationCommand class within Eris...

**Comment by bsian03** [comment]
Am redoing how application commands are handled so don't touch for now pls

**Review by bsian03 — APPROVED** [review]

**Pull request merged**